cloud/jenkins/create_openshift_cluster.groovy

```diff
@@ -0,0 +1,152 @@
+            sh """
+                source $HOME/google-cloud-sdk/path.bash.inc
+                export KUBECONFIG=$WORKSPACE/openshift/$CLUSTER_SUFFIX/auth/kubeconfig
+                for namespace in \$(kubectl get namespaces --no-headers | awk '{print \$1}' | grep -vE "^kube-|^openshift" | sed '/-operator/ s/^/1-/' | sort | sed 's/^1-//'); do
+                    kubectl delete deployments --all -n \$namespace --force --grace-period=0 || true
+                    kubectl delete sts --all -n \$namespace --force --grace-period=0 || true
+                    kubectl delete replicasets --all -n \$namespace --force --grace-period=0 || true
+                    kubectl delete poddisruptionbudget --all -n \$namespace --force --grace-period=0 || true
+                    kubectl delete services --all -n \$namespace --force --grace-period=0 || true
+                    kubectl delete pods --all -n \$namespace --force --grace-period=0 || true
+                done
+                kubectl get svc --all-namespaces || true
+
+                /usr/local/bin/openshift-install destroy cluster --dir=./openshift/$CLUSTER_SUFFIX || true
+            """
+        }
+    }
+}
+
+pipeline {
+    parameters {
+        string(
+            defaultValue: '4.11.41',
+            description: 'OpenShift version to use',
+            name: 'PLATFORM_VER')
+        string(
+            defaultValue: '3',
+            description: 'Number of hours to keep the cluster alive',
+            name: 'CLUSTER_DURATION')
+    }
+    agent {
+         label 'docker'
+    }
+    options {
+        buildDiscarder(logRotator(daysToKeepStr: '-1', artifactDaysToKeepStr: '-1', numToKeepStr: '20', artifactNumToKeepStr: '20'))
+        skipDefaultCheckout()
+    }
+
+    stages {
+        stage('Prepare') {
+            steps {
+                sh '''
+                    if [ ! -d $HOME/google-cloud-sdk/bin ]; then
+                        rm -rf $HOME/google-cloud-sdk
+                        curl https://sdk.cloud.google.com | bash
+                    fi
+
+                    source $HOME/google-cloud-sdk/path.bash.inc
+                    gcloud components update kubectl
+                    gcloud version
+
+                    curl -s -L https://mirror.openshift.com/pub/openshift-v4/clients/ocp/$PLATFORM_VER/openshift-client-linux-$PLATFORM_VER.tar.gz \
+                        | sudo tar -C /usr/local/bin --wildcards -zxvpf -
+                    curl -s -L https://mirror.openshift.com/pub/openshift-v4/clients/ocp/$PLATFORM_VER/openshift-install-linux-$PLATFORM_VER.tar.gz \
+                        | sudo tar -C /usr/local/bin  --wildcards -zxvpf -
+                '''
+            }
+        }
+        stage('Create cluster') {
+            parallel {
+                stage('cluster1') {
+                    steps {
+                        createCluster('cluster1')
+                        sleep(time:"${params.CLUSTER_DURATION}",unit:"HOURS")
```

Comment:
What if the spot instance dies? Who will remove the cluster?

Reply:
nobody, just like in our normal tests

Reply:
as a solution we can use jenkins master for running these jobs or specify non spot instances (pricier)

Reply:
@tplavcic The difference is you can run it for 12 hours and spot will have more time to die. I suggest to have  https://cloud.cd.percona.com/job/pxc-operator-aws-remove-openshift  officially